cumulusci/tasks/metadata_etl/tests/test_base.py

```diff
@@ -379,3 +381,173 @@ def test_transform__xml_parse_error(self):
+                    "entity": "CustomObject",
+                    "tag": "customObjectAttribute",
+                    "value": "newAttributeValue",
+                },
+                "newAttributeValue",
+            ),
+            (
+                {
+                    "managed": False,
+                    "namespace_inject": None,
+                    "entity": "CustomObject",
+                    "tag": "customObjectAttribute",
+                    "value": "%%%NAMESPACE%%%newAttributeValue",
+                },
+                "newAttributeValue",
+            ),
+            (
+                {
+                    "managed": True,
+                    "namespace_inject": None,
+                    "entity": "CustomObject",
+                    "tag": "customObjectAttribute",
+                    "value": "%%%NAMESPACE%%%newAttributeValue",
+                },
+                "newAttributeValue",
+            ),
+            (
+                {
+                    "managed": False,
+                    "namespace_inject": "namespace",
+                    "entity": "CustomObject",
+                    "tag": "customObjectAttribute",
+                    "value": "%%%NAMESPACE%%%newAttributeValue",
+                },
+                "newAttributeValue",
+            ),
+            (
+                {
+                    "managed": True,
+                    "namespace_inject": "namespace",
+                    "entity": "CustomObject",
+                    "tag": "customObjectAttribute",
+                    "value": "%%%NAMESPACE%%%newAttributeValue",
+                },
+                "namespace__newAttributeValue",
+            ),
+        ]:
+            task = create_task(UpdateMetadataFirstChildTextTask, options)
+
+            assert options["entity"] == task.entity
+            assert expected_value == task.options["value"]
+
+    def test_transform_entity__attribute_found(self):
+        api_name = "Supercalifragilisticexpialidocious__c"
+
+        metadata = mock.Mock(spec=MetadataElement)
+        metadata.find.return_value = mock.Mock(spec=MetadataElement)
+        metadata.append.return_value = mock.Mock(spec=MetadataElement)
+
+        entity = "CustomObject"
+        tag = "customObjectAttribute"
+        value = "newAttributeValue"
+
+        task = create_task(
+            UpdateMetadataFirstChildTextTask,
+            {
+                "managed": False,
+                "namespace_inject": None,
+                "entity": entity,
+                "tag": tag,
+                "value": value,
+            },
+        )
+        task.logger = mock.Mock()
+        assert tag == task.options.get("tag")
+        assert value == task.options.get("value")
+
+        actual = task._transform_entity(metadata, api_name)
+
+        assert metadata == actual
+
+        metadata.find.assert_called_once_with(tag)
+        metadata.append.assert_not_called()
+        assert metadata.find.return_value.text == value
```

Comment:
I would suggest not using mocks and treating this as more of a black box test using an actual MetadataElement instance:
1. Construct a MetadataElement tree for the initial XML (possibly using metadata_tree.parse so the XML is readable in the test)
2. Run it through _transform_entity
3. Assert that the result is as expected (possibly after using .tostring() so the expected result is readable in the test)

The point of the test is to make sure the transform made the desired change; we don't need to verify that it was done in a particular way. Also, using mocks means that if we changed something about the real MetadataElement's implementation of these methods that broke _transform_entity, this test would not catch it.

Reply:
I will add a functional test.

Reply:
Done